test/test-jsdomparser.js

```diff
@@ -0,0 +1,194 @@
+
+var baseDoc = new JSDOMParser().parse(BASETESTCASE);
+
+describe("Test JSDOM functionality", function() {
+  it("should work for basic operations with elements", function(done) {
+    expect(baseDoc.childNodes.length).eql(1);
+    expect(baseDoc.getElementsByTagName("*").length).eql(10);
+    var foo = baseDoc.getElementById("foo");
+    expect(foo.id).eql(foo.getAttribute("id"));
+    expect(foo.parentNode.localName).eql("body");
+    expect(baseDoc.body).eql(foo.parentNode);
+    expect(baseDoc.body.parentNode).eql(baseDoc.documentElement);
+    expect(baseDoc.body.childNodes.length).eql(3);
+
+    var generatedHTML = "<html>" + baseDoc.documentElement.innerHTML + "</html>";
+    expect(generatedHTML).eql(BASETESTCASE);
+
+    // Check our script parsing worked:
+    var scripts = baseDoc.getElementsByTagName("script");
+    expect(scripts.length).eql(1);
+    expect(scripts[0].textContent).eql("With < fancy \" characters in it because");
+
+    expect(foo.previousSibling.nextSibling).eql(foo);
+    expect(foo.nextSibling.previousSibling).eql(foo);
+    expect(foo.nextSibling).eql(foo.nextElementSibling);
+    expect(foo.previousSibling).eql(foo.previousElementSibling);
+
+    var beforeFoo = foo.previousSibling;
+    var afterFoo = foo.nextSibling;
+
+    expect(baseDoc.body.lastChild).eql(afterFoo);
+    expect(baseDoc.body.firstChild).eql(beforeFoo);
+
+    var removedFoo = foo.parentNode.removeChild(foo);
+    expect(foo).eql(removedFoo);
+    expect(foo.parentNode).eql(null);
+    expect(foo.previousSibling).eql(null);
+    expect(foo.nextSibling).eql(null);
+    expect(foo.previousElementSibling).eql(null);
+    expect(foo.nextElementSibling).eql(null);
+
+    expect(beforeFoo.localName).eql("p");
+    expect(beforeFoo.nextSibling).eql(afterFoo);
+    expect(afterFoo.previousSibling).eql(beforeFoo);
+    expect(beforeFoo.nextElementSibling).eql(afterFoo);
+    expect(afterFoo.previousElementSibling).eql(beforeFoo);
+
+    expect(baseDoc.body.childNodes.length).eql(2);
+
+    baseDoc.body.appendChild(foo);
+
+    expect(baseDoc.body.childNodes.length).eql(3);
+    expect(afterFoo.nextSibling).eql(foo);
+    expect(foo.previousSibling).eql(afterFoo);
+    expect(afterFoo.nextElementSibling).eql(foo);
+    expect(foo.previousElementSibling).eql(afterFoo);
+
+    // This should reorder back to sanity:
+    baseDoc.body.appendChild(afterFoo);
+    expect(foo.previousSibling).eql(beforeFoo);
+    expect(foo.nextSibling).eql(afterFoo);
+    expect(foo.previousElementSibling).eql(beforeFoo);
+    expect(foo.nextElementSibling).eql(afterFoo);
+
+    expect(foo.previousSibling.nextSibling).eql(foo);
+    expect(foo.nextSibling.previousSibling).eql(foo);
+    expect(foo.nextSibling).eql(foo.nextElementSibling);
+    expect(foo.previousSibling).eql(foo.previousElementSibling);
+    done();
```

Comment:
As all operations within this test are synchronous, using `done` isn't required.